### DISCUSSION TITLE
emojize & symolize - api is not intuitive

### DISCUSSION THREAD

**Discussion by bilal-fazlani** [swiftbar/SwiftBar#113] [Ideas]
This is the documentation

Parameter | Value | Description
-- | -- | --
emojize | False | Disables parsing of GitHub style emojis, i.e. :mushroom: into 🍄. Requires symbolize=false
symbolize | False | Disables parsing of SF Symbols i.e. "SF Symbols Test :sun.max: :cloud.fill: :gamecontroller.fill: 🔖 :sun.dust:"

This is my script

```bash
#!/bin/bash

echo "hello
---
hasdas :mushroom: sadsad | symbolize=false
hi:mushroom:xzczxc | symbolize=false
hasdas :mushroom: sadsad | emojize=true
hi:mushroom:xzczxc | emojize=true
"
```

Below plugin is rendered

![image](https://user-images.githubusercontent.com/3396271/103171170-f7bf2100-486f-11eb-9940-d221edebda7d.png)

The first two items have symbolize set to false and rest of them have emojize set to true. 
The documentation says `"Requires symbolize=false"` but it's a bit confusing (to me as a plugin developer).

Can emojize and symbolize be decoupled?

How about we keep `emojize` as is, but give a different syntax to `symbolize`?

**Comment by melonamin** [comment]
Hey, I'm not sure where the confusion is coming from, let me illustrate:
```
#!/bin/bash
echo "SF Symbol :house:|"
echo "---"
echo "Emoji :house:| symbolize=false"
echo "Text :house:| symbolize=false emojize=false"
```
<img width="223" alt="Screen Shot 2020-12-27 at 9 41 52 AM" src="https://user-images.githubusercontent.com/222100/103173262-c1f64a00-4827-11eb-83d4-15284c06669f.png">

Principles:
- emojis and symbolize have the same syntax, it is cleaner in my eyes
- symbolize has a priority, i.e. if the same literal, like `:house:` in example above, is both an Emoji and SF Symbol - SwiftBar will use SF Symbol
- `symbolize=false` turns off SF Symbols, allowing Emoji conversion

**Comment by bilal-fazlani** [comment]
This helps a lot. Thank you. 
I think I know where my confusion came from. I am using Catalina.  And SF Symbols are not available.

So the same script

```bash
#!/bin/bash
echo "SF Symbol :house:|"
echo "---"
echo "Emoji :house:| symbolize=false"
echo "Text :house:| symbolize=false emojize=false"
```

appears like this on my machine

![image](https://user-images.githubusercontent.com/3396271/103173915-06173800-4884-11eb-9358-783cf4e4d874.png)

So in order to fix this on Catalina, I have to modify my script to be like:

```bash
#!/bin/bash

echo "SF Symbol :house: | symbolize=false"
echo "---"
echo "Emoji :house:| symbolize=false"
echo "Text :house:| symbolize=false emojize=false"
```

This make it render like 

![image](https://user-images.githubusercontent.com/3396271/103174007-a66d5c80-4884-11eb-9c7e-c5a9e0a17c33.png)

The only drawback of explicitly setting `symbolize=false` for backward compatibility for plugin is that, it renders emojis on Big Sur too.

If SwiftBar automatically assumed `symbolize=false` on pre BigSur OS, then,  this

```bash
#!/bin/bash
echo "SF Symbol :house:|"
echo "---"
echo "Emoji :house:| symbolize=false"
echo "Text :house:| symbolize=false emojize=false"
```

would result into 
 ![image](https://user-images.githubusercontent.com/3396271/103174007-a66d5c80-4884-11eb-9c7e-c5a9e0a17c33.png)
 on catalina 

and the same script would result into rendering SF Symbol on BigSur

![image](https://user-images.githubusercontent.com/222100/103173262-c1f64a00-4827-11eb-83d4-15284c06669f.png)

**Comment by bilal-fazlani** [comment]
Developers can of course opt out of this by explicitly setting symbolize= false

**Comment by melonamin** [comment]
> If SwiftBar automatically assumed symbolize=false on pre BigSur OS, then, this

That makes sense, implemented in #114

**Reply by bilal-fazlani** [reply]
Thank you!

**Comment by myelsukov** [comment]
QQ: How can I use **SF Symbols** with Catalina (10.15.7)? I have installed **SF Symbols 2** but whatever I try I can't use them with SwiftBar.

**Reply by melonamin** [reply]
Per Apple’s HIG: “SF Symbols are available in iOS 13 and later, macOS 11 and later, watchOS 6 and later, and tvOS 13 and later.”

I’m not sure why you are able to install SF Symbols app, but SF Symbols are not supported in Catalina 😔
